packages/core/src/routes/types.ts

```diff
@@ -3,10 +3,11 @@ import Router from 'koa-router';
 import { WithAuthContext } from '@/middleware/koa-auth';
 import { WithI18nContext } from '@/middleware/koa-i18next';
 import { WithLogContext } from '@/middleware/koa-log';
+import { WithOidcInteractionContext } from '@/middleware/koa-oidc-interaction';
 import { WithUserInfoContext } from '@/middleware/koa-user-info';
 
-export type AnonymousRouter = Router<unknown, WithLogContext<WithI18nContext>>;
-export type AuthedRouter = Router<
-  unknown,
-  WithUserInfoContext<WithAuthContext<WithLogContext<WithI18nContext>>>
->;
+type AnonymousContext = WithLogContext<WithI18nContext>;
+
+export type AnonymousRouter = Router<any, AnonymousContext>;
+export type AuthedRouter = Router<any, WithUserInfoContext<WithAuthContext<AnonymousContext>>>;
+export type SessionRouter = Router<any, WithOidcInteractionContext<AnonymousContext>>;
```

Comment:
should be Router<unknown, ContextT>?

Reply:
cannot. `ContextT` does not exist. (just like generic type `T`)

Reply:
They are used to infer `ctx` containing which fields.

![image](https://user-images.githubusercontent.com/10594507/167343035-4173417e-2ec5-4b16-8d45-d3c4fe536545.png)

Reply:
fixed. PTAL